CONTRIBUTING.md

```diff
@@ -40,177 +264,782 @@ since someone else can often do this if you include a complete example. We
+from your side but nothing happens for a few days).
+
+After your PR is accepted and merged, it may get backported to a
+stable branch if relevant, and will eventually make it to a release.
+You do not have to worry about this, it is the role of the assignee
+and the release manager to do so.  The milestone should give you an
+indication of when to expect your change to be released (this could be
+several months after your PR is merged).  That said, you can start
+using the latest Coq `master` branch to take advantage of all the new
+features, improvements, and fixes.
+
+### Taking feedback into account ###
+
+#### Understanding automatic feedback ####
+
+When you open or update a PR, you get automatically some feedback: we
+have a bot whose job will be to push a branch to our GitLab mirror to
+run some continuous integration (CI) tests.  The tests will run on a
+commit merging your branch with the base branch, so if there is a
+conflict and this merge cannot be performed automatically, the bot
+will put a `needs: rebase` label, and the tests won't run.
+
+Otherwise, a large suite of tests will be run on GitLab, plus some
+additional tests on Azure for Windows and macOS compatibility.
+
+If a test fails on GitLab, you will see in the GitHub PR interface,
+both the failure of the whole pipeline, and of the specific failed
+job.  Most of these failures indicate problems that should be
+addressed, but some can still be due to synchronization issues out of
+your control.  In particular, if you get a failure in one of the
+tested plugins but you didn't change the Coq API, it is probably a
+transient issue and you shouldn't have to worry about it.  In case of
+doubt, ask the reviewers.
+
+##### Test-suite failures #####
+
+If you broke the test-suite, you should get many failed jobs, because
+the test-suite is run multiple times in various settings.  You should
+get the same failure locally by running `make test-suite` or `make -f
+Makefile.dune test-suite`.  It's helpful to run this locally and
+ensure the test-suite is not broken before submitting a PR as this
+will spare a lot of runtime on distant machines.
+
+To learn more about the test-suite, you should refer to its
+[README][test-suite-README].
+
+##### Linter failures #####
+
+We have a linter that checks a few different things:
+
+- **Every commit can build.** This is an important requirement to
+  allow the use of `git bisect` in the future.  It should be possible
+  to build every commit, and in principle even the test-suite should
+  pass on every commit (but this isn't tested in CI because it would
+  take too long).  You can test and fix this locally with `git
+  rebase`.
+- **No tabs or end-of-line spaces on updated lines** (and all files
+  should end with a single newline).  We are trying to get rid of all
+  tabs (except in the `Makefile`s) and all end-of-line spaces from the
+  code base.  This checks not only that you didn't introduce new ones,
```

Comment:
We should maybe have a link here to "coding style" and just say "Coq files must not use tabs for indentation"